addon/index.d.ts

```diff
@@ -0,0 +1,759 @@
+ * via {@linkcode TaskInstance#cancel} or {@linkcode Task#cancelAll},
+ * or automatically due to the host object being destroyed, or
+ * because concurrency policy enforced by a
+ * {@linkcode TaskProperty Task Modifier} canceled the task instance.
+ */
+export interface TaskInstance<T = unknown> extends Promise<T> {
+  /**
+   * If this TaskInstance runs to completion by returning a property
+   * other than a rejecting promise, this property will be set
+   * with that value.
+   */
+  readonly value: T | null;
+
+  /**
+   * If this TaskInstance is canceled or throws an error (or yields
+   * a promise that rejects), this property will be set with that error.
+   * Otherwise, it is null.
+   */
+  readonly error: unknown;
+
+  /**
+   * True if the task instance is fulfilled.
+   */
+  readonly isSuccessful: boolean;
+
+  /**
+   * True if the task instance resolves to a rejection.
+   */
+  readonly isError: boolean;
+
+  /**
+   * True if the task instance was canceled before it could run to completion.
+   */
+  readonly isCanceled: boolean;
+
+  /**
+   * True if the task instance has started, else false.
+   */
+  readonly hasStarted: boolean;
+
+  /**
+   * True if the task has run to completion.
+   */
+  readonly isFinished: boolean;
+
+  /**
+   * True if the task is still running.
+   */
+  readonly isRunning: boolean;
+
+  /**
+   * Describes the state that the task instance is in. Can be used for debugging,
+   * or potentially driving some UI state. Possible values are:
+   *
+   * - `"dropped"`: task instance was canceled before it started
+   * - `"canceled"`: task instance was canceled before it could finish
+   * - `"finished"`: task instance ran to completion (even if an exception was thrown)
+   * - `"running"`: task instance is currently running (returns true even if
+   *     is paused on a yielded promise)
+   * - `"waiting"`: task instance hasn't begun running yet (usually
+   *     because the task is using the {@linkcode TaskProperty#enqueue .enqueue()}
+   *     task modifier)
+   *
+   * The animated timeline examples on the [Task Concurrency](/#/docs/task-concurrency)
+   * docs page make use of this property.
+   */
+  readonly state: 'dropped' | 'canceled' | 'finished' | 'running' | 'waiting';
+
+  /**
+   * True if the TaskInstance was canceled before it could
+   * ever start running. For example, calling
+   * {@linkcode Task#perform .perform()} twice on a
+   * task with the {@linkcode TaskProperty#drop .drop()} modifier applied
+   * will result in the second task instance being dropped.
+   */
+  readonly isDropped: boolean;
+
+  /**
+   * Cancels the task instance. Has no effect if the task instance has
+   * already been canceled or has already finished running.
+   *
+   * @param cancelReason Defaults to `".cancel() was explicitly called"`.
+   */
+  cancel(cancelReason?: string): void;
+
+  /**
+   * Returns a promise that resolves with the value returned
```

Comment:
I think this is wrong, or at least worded confusingly. I kept it because I am trying to copy the docs 1:1.

Reply:
I think keeping it the same is probably fine for now. I'm planning to do a broader revisiting of the docs shortly.